src/lib/stores.ts

```diff
@@ -50,29 +50,35 @@ function createSns() {
 			set({
 				name: '',
 				principal: '',
-				encodedAccount: '-/-',
+				encodedAccount: undefined,
 				icpBalance: undefined,
 				nicpBalance: undefined
 			})
 	};
 }
-export const sns = createSns();
+export const sns = createBoomerangSnsStore();
 
-export async function handleSnsChange(name?: string, principal?: string) {
-	const boomerang = get(canisters)?.boomerang;
-	if (!boomerang) return;
+export const handleSnsChange = async (name?: string, principal?: string) => {
```

Comment:
Where we calling that function with empty args?

Reply:
Yess when initializing custom Sns